File: utils/read_cert.py
import sys

with open(sys.argv[1]) as f:
    lines = f.read().splitlines()
    if len(lines) == 1:
        s = lines[0]
        lines = [s[i:i+64] for i in range(0, len(s), 64)]
        lines.insert(0, "-----BEGIN CERTIFICATE-----")
        lines.append("-----END CERTIFICATE-----")
        print("\\n".join(lines))
    else:
        print("\\n".join(lines))

File: t/lib/read_cert.lua
local _M = {}

local function split(text, chunk_size)
    local s = {}
    for i=1, #text, chunk_size do
        s[#s+1] = text:sub(i, i + chunk_size - 1)
    end
    return s
end

function _M.read_cert(str)
    local t = split(str, 64)
    table.insert(t, 1, "-----BEGIN CERTIFICATE-----")
    table.insert(t, "-----END CERTIFICATE-----")
    return string.format(table.concat(t, "\n"))
end

local function read_whole_file(file)
    local f = assert(io.open(file, "rb"))
    local content = f:read("*all")
    f:close()
    return content
end

function _M.read_cert_file(file)
    return _M.read_cert(read_whole_file(file))
end

return _M
